pkg/kubectl/autoscale.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"strconv"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+const (
+	DefaultCPUUtil   = 80
+	scaleSubResource = "scale"
+)
+
+type HorizontalPodAutoscalerV1Beta1 struct{}
+
+func (HorizontalPodAutoscalerV1Beta1) ParamNames() []GeneratorParam {
+	return []GeneratorParam{
+		{"default-name", true},
+		{"name", false},
+		{"scaleRef-kind", false},
+		{"scaleRef-namespace", false},
+		{"scaleRef-name", false},
+		{"scaleRef-apiVersion", false},
+		{"scaleRef-subresource", false},
+		{"min", true},
+		{"max", true},
+		{"cpu%", false},
```

Comment:
We should probably have a note in https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#flag-conventions about not using non-alphanumeric characters in flag names -- I suspect people would just be confused by that.

I do like the idea of something like `--percent=k1=v1,k2=v2`, although perhaps it could be `--target=cpu=50%,foo=30` to allow for different metrics in the future more easily (e.g. custom metrics where percentage is not a relevant way to measure things)